O3-1.tex

```diff
@@ -279,14 +279,12 @@ \section{The LSST software stack}
 extraction of scientifically useful characteristics of those objects.
 While firmly footed in this prior history, the LSST software stack has
 largely been written anew, for reasons of performance, extendability, and
-maintainability. All LSST codes have been designed and implemented
+maintainability. All LSST codes are being developed with the intent of
 following software engineering best practices, including modularity, clear definition
 of interfaces, continuous integration,
 utilization of unit testing, a single set of documentation and coding
-standards, and others. The primary implementation language is Python and, where
-necessary for performance reasons, C++\footnote{All components implemented
-in C++ have been wrapped and exposed as Python modules to the rest
-of the system. Python truly is the ``native language of LSST''.}.
+standards, and others. The primary user-facing implementation language is Python and, where
+necessary for performance reasons or when the algorithms require access to complex data structures, C++.
```

Comment:
Why not?  The reason that object migrate into C++ is to avoid calling out to Python objects in inner loops of C++ algorithms; that's a computational performance issue.